dotnet/Gherkin/AstNode.cs

```diff
@@ -11,86 +15,90 @@ public Token GetToken(TokenType tokenType)
         return GetSingle<Token>((RuleType)tokenType);
     }
 
-    public IEnumerable<Token> GetTokens(TokenType tokenType)
+    public Enumerable<Token> GetTokens(TokenType tokenType)
     {
         return GetItems<Token>((RuleType)tokenType);
     }
 
     public T GetSingle<T>(RuleType ruleType)
     {
-        if (!subItems.TryGetValue(ruleType, out var items))
-            return default;
-        if (items is List<object> list)
+        bool foundOne = false;
+        T ret = default;
+        foreach ((var itemType, var item) in subItems)
         {
-            T ret = default;
-            bool foundOne = false;
-            foreach (var item in list)
-            {
-                if (item is T tItem)
-                {
-                    if (foundOne)
-                        throw new InvalidOperationException();
-                    ret = tItem;
-                    foundOne = true;
-                }
-            }
+            if (itemType != ruleType)
+                continue;
+            if (item is not T tItem)
+                continue;
             if (foundOne)
-                return ret;
-            else
                 throw new InvalidOperationException();
+            ret = tItem;
+            foundOne = true;
         }
-        else if (items is T tItem)
-        {
-            return tItem;
-        }
-        return default;
+        return ret;
     }
 
-    public IEnumerable<T> GetItems<T>(RuleType ruleType)
+    public readonly struct Enumerable<T> : IEnumerable<T>

```

Comment:
I know it is a generic type, but could we find a little bit more specific name, like `ItemsEnumerable`?

Reply:
I changed the name 🙂